src/components/search/data/hooks.js

```diff
@@ -107,19 +108,48 @@ export const useNbHitsFromSearchResults = (searchResults) => {
   return nbHits;
 };
 
-export const useDefaultSearchFilters = ({ enterpriseConfig, subscriptionPlan }) => {
+export const getCatalogString = (offerCatalogs, initialString = '') => {
+  const offerCatalogsCopy = [...offerCatalogs];
+  const lastCatalog = offerCatalogsCopy.pop();
+  return `${offerCatalogsCopy.reduce(
+    (result, catalog) => `${result}enterprise_catalog_uuids:${catalog} OR `, initialString,
+  )}enterprise_catalog_uuids:${lastCatalog}`;
+};
+
+export const useDefaultSearchFilters = ({ enterpriseConfig, subscriptionPlan, offerCatalogs = [] }) => {
+  // default to showing all catalogs
+  const { showAllCatalogs, setShowAllCatalogs } = useContext(SearchContext);
+
+  useMemo(() => {
+    // if there are no subscriptions or offers, we default to showing all catalogs
+    if (!subscriptionPlan && offerCatalogs.length < 1) {
+      setShowAllCatalogs(true);
+    }
+  }, [subscriptionPlan, offerCatalogs.length]);
+
   const filters = useMemo(
     () => {
+      if (showAllCatalogs) {
+        if (subscriptionPlan) {
+          // show subscription catalog and all other enterprise catalogs
+          return `enterprise_catalog_uuids:${subscriptionPlan.enterpriseCatalogUuid} OR enterprise_customer_uuids:${enterpriseConfig.uuid}`;
+        }
+        // show all enterprise catalogs
+        return `enterprise_customer_uuids:${enterpriseConfig.uuid}`;
+      }
       // if there's a subscriptionPlan, filter results by the subscription catalog
+      // and any catalogs for which the user has vouchers
       if (subscriptionPlan) {
+        if (offerCatalogs.length > 0) {
+          return `enterprise_catalog_uuids:${subscriptionPlan.enterpriseCatalogUuid} ${getCatalogString(offerCatalogs, 'OR ')}`;
```

Comment:
Rather than combining the subs catalog with the offer catalogs in the filter string, I think we could do something like:

```jsx
if (offerCatalogs.length > 0) {
  const catalogs = [subscriptionPlan.enterpriseCatalogUuid, ...offerCatalogs];
  return getCatalogString(catalogs);
}
```

Reply:
Oh, yeah, I like that.

Reply:
Oh, but it does not work because one is `enterprise_catalog` and the other is `enterprise_customer`